pkg/component/certmanagement/deployment.go

```diff
@@ -0,0 +1,400 @@
+			},
+		)
+		container.VolumeMounts = append(container.VolumeMounts, corev1.VolumeMount{
+			Name:      "ca-certificates",
+			ReadOnly:  true,
+			MountPath: "/var/run/cert-manager/certs",
+		})
+		deployment.Spec.Template.Spec.Volumes = append(deployment.Spec.Template.Spec.Volumes, corev1.Volume{
+			Name: "ca-certificates",
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName: caCertSecret.Name,
+				},
+			},
+		})
+		utilruntime.Must(references.InjectAnnotations(deployment))
+	}
+
+	resources, err := registry.AddAllAndSerialize(objects...)
+	if err != nil {
+		return err
+	}
+
+	return createManagedResource(ctx, d.client, DeploymentManagedResourceName, false, resources)
+}
+
+func (d *certManagementDeployment) deleteHelmRelease(ctx context.Context) error {
+	if err := d.client.DeleteAllOf(ctx, &corev1.Secret{}, client.InNamespace(d.namespace), client.MatchingLabels{
+		"name":  "cert-management",
+		"owner": "helm",
+	}); err != nil {
+		return fmt.Errorf("deleting secrets for old Helm releases failed: %w", err)
+	}
+	return nil
+}
+
+func (d *certManagementDeployment) Destroy(ctx context.Context) error {
+	return deleteManagedResource(ctx, d.client, DeploymentManagedResourceName)
+}
+
+// TimeoutWaitForManagedResource is the timeout used while waiting for the ManagedResources to become healthy
+// or deleted.
+var TimeoutWaitForManagedResource = 2 * time.Minute
+
+func (d *certManagementDeployment) Wait(ctx context.Context) error {
+	timeoutCtx, cancel := context.WithTimeout(ctx, TimeoutWaitForManagedResource)
+	defer cancel()
+
+	return managedresources.WaitUntilHealthy(timeoutCtx, d.client, v1beta1constants.GardenNamespace, DeploymentManagedResourceName)
+}
+
+func (d *certManagementDeployment) WaitCleanup(ctx context.Context) error {
+	timeoutCtx, cancel := context.WithTimeout(ctx, TimeoutWaitForManagedResource)
+	defer cancel()
+
+	return managedresources.WaitUntilDeleted(timeoutCtx, d.client, v1beta1constants.GardenNamespace, DeploymentManagedResourceName)
+}
+
+func getDeploymentLabels() map[string]string {
+	return map[string]string{
+		appName:     componentName,
+		appInstance: componentName,
+	}
+}
+
+func getObjectKeyLocalObjectRef(ref corev1.LocalObjectReference) client.ObjectKey {
+	return client.ObjectKey{Namespace: v1beta1constants.GardenNamespace, Name: ref.Name}
+}
+
+func createManagedResource(ctx context.Context, client client.Client, name string, keepObjects bool, data map[string][]byte) error {
+	return managedresources.Create(ctx, client, v1beta1constants.GardenNamespace, name, map[string]string{appName: componentName},
+		true, v1beta1constants.SeedResourceManagerClass, data, &keepObjects, map[string]string{appName: componentName}, nil)
+}
+
+func deleteManagedResource(ctx context.Context, client client.Client, name string) error {
+	return managedresources.Delete(ctx, client, v1beta1constants.GardenNamespace, name, true)
```

Comment:
There is a [DeleteForSeed](https://github.com/gardener/gardener/blob/d30f7f5a57a86dc970d4700b4c62e79590a01fcc/pkg/utils/managedresources/managedresources.go#L280C2-L280C15) function for this case.

Reply:
replaced